lib/client/rpcClient.ts

```diff
@@ -0,0 +1,58 @@
+import { Program, Provider, utils } from "@coral-xyz/anchor";
+import { AutocratProgram, Dao, ProgramVersion } from "../types";
+import { Metaplex } from "@metaplex-foundation/js";
+import { TokenStandard } from "@metaplex-foundation/mpl-token-metadata";
+import { PublicKey } from "@solana/web3.js";
+import { MetaDAOClient } from "./client";
+
+export class MetaDAORPClient implements MetaDAOClient {
+  private programVersion: ProgramVersion;
+  private rpcProvider: Provider;
+  private metaplex: Metaplex;
+  private constructor(programVersion: ProgramVersion, rpcProvider: Provider) {
+    this.programVersion = programVersion;
+    this.rpcProvider = rpcProvider;
+    this.metaplex = Metaplex.make(this.rpcProvider.connection);
+  }
+  static make(programVersion: ProgramVersion, rpcProvider: Provider) {
+    return new MetaDAORPClient(programVersion, rpcProvider);
+  }
+  async fetchDao(daoAddress: string): Promise<Dao | undefined> {
+    const { programId, idl } = this.programVersion!;
+    const dao = PublicKey.findProgramAddressSync(
+      [utils.bytes.utf8.encode(daoAddress)],
+      programId
+    )[0];
+
+    const daoTreasury = PublicKey.findProgramAddressSync(
+      [dao.toBuffer()],
+      programId
+    )[0];
+
+    const autocratProgram = new Program<AutocratProgram>(
+      idl as AutocratProgram,
+      programId,
+      this.rpcProvider
+    );
+    const daoState = await autocratProgram.account.dao.fetch(dao);
+    const baseToken = await this.metaplex
```

Comment:
So we can cut this. Behind the scenes metaplex does alllll kinds of things to get the data. https://github.com/metaplex-foundation/js/blob/main/packages/js/src/plugins/nftModule/operations/findNftByMint.ts

```typescript
// old version
const baseTokenMint = daoState.metaMint
const baseTokenMint = daoState.tokenMint

const baseToken = await connection.getAccountInfo(baseTokenMint)
```

Which should return something like:

```json
"value":{"data":{"parsed":{"info":{"decimals":6,"freezeAuthority":"3sNBr7kMccME5D55xNgsmYpZnzPgP2g12CixAajXypn6","isInitialized":true,"mintAuthority":"2wmVCSfPxGPjrnMMn7rchp4uaeoTqN39mXFC2zhPdri9","supply":"2514620541542759"},"type":"mint"},"program":"spl-token","space":82},"executable":false,"lamports":242943372956,"owner":"TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA","rentEpoch":18446744073709551615,"space":82}}
```
This will get us basically decimals only. Isn't ideal.
Now there is a caveat here, in that we don't have a name for it. There are a few options in that we can use metaplex, we can create a mapping table, we can hardcode and map from the table, or we can make some other calls.

Given the number of tokens scales with the customers, having a strict list is most likely fine. We just don't need to fetch this data once we have it, so fetching it seems like wasted time. We know from the dao what the token and usdc are, so why fight it? Index or just put a list together somewhere we maintain.

Reply:
@R-K-H I guess you are saying every time we add a new customer we have to update the SDK/frontend with the new token? Or should we assemble a massive json of all the DAOs and their token's from the start? Idk... it's not an idea I'm crazy about because it immediately does not scale. Would love to chat about it if you have time.

Reply:
The new implementation for token metadata/fetching is in when you are ready to take a look.